knotx-knots/knotx-knot-handlebars/src/main/java/com/cognifide/knotx/knot/templating/HandlebarsFragment.java

```diff
@@ -15,54 +15,56 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.cognifide.knotx.knot.templating.wrappers;
+package com.cognifide.knotx.knot.templating;
 
 import com.cognifide.knotx.fragments.Fragment;
 import com.cognifide.knotx.handlebars.JsonObjectValueResolver;
 import com.github.jknack.handlebars.Context;
 import com.github.jknack.handlebars.Handlebars;
 import com.github.jknack.handlebars.Template;
-
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.io.IOException;
+import org.apache.commons.lang3.StringUtils;
 import org.jsoup.Jsoup;
 import org.jsoup.nodes.Document;
 import org.jsoup.nodes.Element;
 
-import java.io.IOException;
+class HandlebarsFragment {
 
-import io.vertx.core.json.JsonObject;
+  private static final Logger LOGGER = LoggerFactory.getLogger(HandlebarsFragment.class);
 
-public class TemplateHtmlFragment implements HtmlFragment {
+  private static final int MAX_FRAGMENT_CONTENT_LOG_LENGTH = 256;
 
-  private final String unwrappedContent;
   private final Fragment fragment;
-  private final JsonObject context;
 
-  private Template compiledFragment;
+  private final String unwrappedContent;
 
-  public TemplateHtmlFragment(Fragment fragment) {
+  HandlebarsFragment(Fragment fragment) {
     this.fragment = fragment;
-    this.context = fragment.getContext();
-    Document document = Jsoup.parseBodyFragment(fragment.getContent());
-    Element scriptTag = document.body().child(0);
-    this.unwrappedContent = scriptTag.unwrap().toString(); //remove outer script tag
+    this.unwrappedContent = getUnwrappedContent(fragment);
   }
 
-  @Override
-  public String getContentWithContext(JsonObject model) {
+  String compileWith(Handlebars handlebars) {
     try {
-      return compiledFragment.apply(Context.newBuilder(model.mergeIn(context)).push(JsonObjectValueResolver.INSTANCE).build());
+      Template compiledFragment = handlebars.compileInline(unwrappedContent);
+      LOGGER.trace("Applying context [{}] to fragment [{}]", fragment.getContext(),
+          StringUtils.substring(fragment.getContent(), 0, MAX_FRAGMENT_CONTENT_LOG_LENGTH)
```

Comment:
instead of substring use abbreviate method from stringUtils
```java
StringUtils.replaceAll("[\n\r\t]","").abbreviate(fragment.getContent(), MAX_FRAGMENT_CONTENT_LOG_LENGTH);
```
This will also add ellipsis at the end `...`  (and replace `\t` - tabs to get more content visible..)